src/pyedb/dotnet/edb_core/cell/hierarchy/component.py

```diff
@@ -828,23 +830,30 @@ def assign_spice_model(self, file_path, name=None, sub_circuit_name=None):
         with open(file_path, "r") as f:
             for line in f:
                 if "subckt" in line.lower():
-                    pinNames = [i.strip() for i in re.split(" |\t", line) if i]
-                    pinNames.remove(pinNames[0])
-                    pinNames.remove(pinNames[0])
+                    pin_names_sp = [i.strip() for i in re.split(" |\t", line) if i]
+                    pin_names_sp.remove(pin_names_sp[0])
+                    pin_names_sp.remove(pin_names_sp[0])
                     break
-        if len(pinNames) == self.numpins:
-            model = self._edb.cell.hierarchy._hierarchy.SPICEModel()
-            model.SetModelPath(file_path)
-            model.SetModelName(name)
-            if sub_circuit_name:
-                model.SetSubCkt(sub_circuit_name)
-            terminal = 1
-            for pn in pinNames:
-                model.AddTerminalPinPair(pn, str(terminal))
-                terminal += 1
-        else:  # pragma: no cover
-            logging.error("Wrong number of Pins")
-            return False
+
+        model = self._edb.cell.hierarchy._hierarchy.SPICEModel()
+        model.SetModelPath(file_path)
+        model.SetModelName(name)
+        if sub_circuit_name:
+            model.SetSubCkt(sub_circuit_name)
+
+        if not len(pin_names_sp) == self.numpins:  # pragma: no cover
+            raise ValueError(f"Pin counts doesn't match component {self.name}.")
```

Comment:
This should be moved before defining model, right ?

Reply:
Good point